source/routing/isochrone.cpp

```diff
@@ -161,6 +161,39 @@ struct InfoCircle {
         duration_left(duration_left) {}
 };
 
+static bool within_info_circle(const InfoCircle& circle,
+                              const std::vector<InfoCircle>& multi_poly,
+                              const double& speed) {
+    if (multi_poly.empty()) {return false;}
+    auto begin = multi_poly.begin();
+    const auto end = multi_poly.end();
+    double circle_radius = circle.duration_left * speed;
+    double coslat = cos(circle.center.lat() * type::GeographicalCoord::N_DEG_TO_RAD);
+    return any_of(begin, end, [speed, circle, circle_radius, coslat](const InfoCircle& it) {
+        double it_radius = it.duration_left * speed;
+        return sqrt(circle.center.approx_sqr_distance(it.center, coslat)) + circle_radius < it_radius;
+    });
+}
+
+static std::vector<InfoCircle> delete_useless_circle(std::vector<InfoCircle> circles,
+                                                     const double& speed) {
+    std::vector<InfoCircle> circles_utils;
+    if (!circles.empty()) {
+        boost::sort(circles,
+                    [](const InfoCircle& a, const InfoCircle& b) {return a.duration_left > b.duration_left;});
+        auto it = circles.begin();
+        circles_utils.push_back(*it);
+        const auto end = circles.end();
+        ++it;
+        for (; it != end; ++it) {
+            if (!within_info_circle(*it, circles_utils, speed)) {
+                circles_utils.push_back(*it);
```

Comment:
i don't know if it'd be better to move instead of copy? `circles_utils.push_back(std::move(*it));`@TeXitoi